libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -661,7 +661,7 @@ int AP_HAL__I2CDevice_transfer(lua_State *L) {
 
     // Parse string of bytes to send
     size_t send_len;
-    const uint8_t* send_data = (const uint8_t*)(lua_tolstring(L, 2, &send_len));
+    const uint8_t* send_data = (const uint8_t*)(luaL_checklstring(L, 2, &send_len));
```

Comment:
Its valid to pass nulptr to `transfer`, it is required by some devices. grep for `transfer(nullptr`.

So it that does work as expected from lua I think its fine to leave this one? The lus would be something like:

`AP_HAL__I2CDevice_ud:transfer(nil, 2)`

If that is safe we should add a comment here explaining why its OK and update the docs that you can pass nil for the first argument to only read.

Reply:
I guess we could add a check that if it not a string it must be exactly nil, so passing other random stuff will error out.

Reply:
Does it matter what the pointer is if the length argument is 0? A user could pass an empty string to get a zero length.

Reply:
Hum, the chibiOS bounce buffer stuff checks for nulprt so it might matter. I guess you could always pass nulptr if the length is zero.

Reply:
I like that option more, I will update to do that.